app/charts/index.ts

```diff
@@ -1013,36 +1020,25 @@ const chartConfigsAdjusters: ChartConfigsAdjusters = {
           });
         },
       },
-      segment: ({
-        oldValue,
-        oldChartConfig,
-        newChartConfig,
-        dimensions,
-        measures,
-      }) => {
+      segment: ({ oldValue, oldChartConfig, newChartConfig, measures }) => {
         let newSegment: ColumnSegmentField | undefined;
-        let newColor: ColorField | undefined;
 
         const yMeasure = measures.find(
           (d) => d.id === newChartConfig.fields.y.componentId
         );
 
         // When switching from a table chart, a whole fields object is passed as oldValue.
         if (oldChartConfig.chartType === "table") {
-          const maybeSegmentAndColorFields =
-            convertTableFieldsToSegmentAndColorFields({
-              fields: oldValue as TableFields,
-              dimensions,
-              measures,
-            });
+          const maybeSegmentField = convertTableFieldsToSegmentFields({
+            fields: oldValue as TableFields,
+          });
```

Comment:
@noahonyejese I am not exactly sure where the code is wrong, but you can see that the colors are set wrongly when going to Photovoltaikanlagen cube, adding segmentation by Kanton, switching to a table chart and back to a column chart – we get all blue bars then 👀 

<img width="1728" alt="Screenshot 2025-02-21 at 09 47 35" src="https://github.com/user-attachments/assets/80c3132e-e998-49f6-8cfd-1fd65bc8800a" />

Reply:
@noahonyejese I checked the same flow as above, and now it looks like the segmentation is not removed when switching to a column chart, and the chart gets broken 🥲 

<img width="2560" alt="Screenshot 2025-02-24 at 09 03 03" src="https://github.com/user-attachments/assets/fbeecb6b-576f-457c-94da-d4972286fa12" />

Reply:
(or it's just the colors that are wrong)